### PR TITLE
v0.26: Filter expressions limited to 2000 concatenated filters

### LINKED ISSUE #1415: v0.26: Filter expressions limited to 2000 concatenated filters
## Summary
The linked issue was opened by a user with an insane amount of chained `OR` filters, which caused a stack overflow crash. One of the solutions proposed by the core team is to limit filter usage so MeiliSearch does not accept/ignores searches with >2000 filters.

EDIT: contrary to our initial understanding, this limit only applies to `OR` filters. If I understood @curquiza's explanation correctly, `OR` filters create nested structures which can lead to the stack overflowing. For example:

1. `user = 1 AND category = "action" AND language = "RU"` -> translates into a flat query
2. `user = 1 OR user = 2 OR user = 3` -> translates into a query nested twice

The new limit would thus be triggered if a programmer were to make a query like  number 2, but with >2000 OR clauses:

`user = 1 OR user = 2 […] OR user = 1500 OR user = 1501 […] OR user = 2000 OR user = 2001`

The same limit also applies to the array syntax:

```
  [
    ["user = 1", "user = 2", etc]
  ]
```

## Tasks
- [ ] Update `learn/advanced/known_limitations.md` with the new filter limitation, following the Limitation/Explanation format used elsewhere on the page

## References
Issue on core: https://github.com/meilisearch/MeiliSearch/issues/1967
PR/line with reference code: https://github.com/meilisearch/milli/pull/421/files#diff-a1253bb15ffe3a75231197ec3f7905987e24a10458dd5255f5fd4a75fbafe192R19
Discussion: https://github.com/meilisearch/documentation/discussions/1365#discussioncomment-1955086

### PR DESCRIPTION
**Pull request by maryamsulemani97**
closes #1415

### CODE REVIEW

**Review by dichotommy — COMMENTED** [review]
This looks fine to me. Maybe at the end of the explanation, we could add a brief sentence and/or example on what constitutes an `OR` filter, since an `OR` filter in array syntax might not be immediately recognizable for users.

I'm unsure of the use of the word "concatenated" here. I think "Maximum filter depth" would be a much clearer title for the section, but maybe that's just me.

File: learn/advanced/known_limitations.md [hunk 1]
```diff
@@ -69,6 +69,26 @@ If your query is `Hello - World`:
 
 **Explanation:** Meilisearch stores `filterableAttributes` values as keys in LMDB, a datatype whose size is limited to approximately 500 bytes. Note that this only applies to individual values—for example, a `genres` attribute can contain any number of values such as `horror`, `comedy`, or `cyberpunk` as long as each one of them is smaller than 500 bytes.
 
+### Maximum filter depth
+
+**Limitation:** Meilisearch does not accept searches with more than 2000 `OR` filters.
+
+**Explanation:** `OR` filters create nested structures which can lead to a stack overflow.
+
+### Example
```
Comment (dichotommy) [inline learn/advanced/known_limitations.md:78]:
```suggestion
#### Example
```

File: learn/advanced/known_limitations.md [hunk 2]
```diff
@@ -69,6 +69,26 @@ If your query is `Hello - World`:
...
+
+If you tried filtering what users have access to a certain index using either of the following queries, they will not work:
```
Comment (dichotommy) [inline learn/advanced/known_limitations.md:80]:
```suggestion
The following filter expressions will cause search queries to fail:
```

**Review by dichotommy — CHANGES_REQUESTED** [review]

**Review by dichotommy — APPROVED** [review]
LGTM! 🤿

File: learn/advanced/known_limitations.md [hunk 3]
```diff
@@ -69,6 +69,26 @@ If your query is `Hello - World`:
...
+
+```
+user = 1 OR user = 2 […] OR user = 1500 OR user = 1501 […] OR user = 2000 OR user = 2001
+```
+
+```
+[
+  ["user = 1", "user = 2", […], "user = 1500", "user = 1501", […], "user = 2000", "user = 2001"]
+]
+```
```
Comment (guimachiavelli) [inline learn/advanced/known_limitations.md:90]: Perhaps making this a json code block would improve reading slightly?

File: learn/advanced/known_limitations.md [hunk 4]
```diff
@@ -69,6 +69,26 @@ If your query is `Hello - World`:
...
+
+**Limitation:** Meilisearch does not accept searches with more than 2000 `OR` filters.
+
+**Explanation:** `OR` filters create nested structures which can lead to a stack overflow.
+
+### Example
+
+Either of these filter expressions would cause a search query to fail:
+
+```json
```
Comment (Kerollmops) [inline learn/advanced/known_limitations.md:82]:
```suggestion
```sql
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Apart from a  possible highlighting problem, it looks good to me.

**Pull request merged**